code/modules/mob/living/simple_animal/simple_animal.dm

```diff
@@ -408,6 +411,28 @@
 	update_action_buttons_icon()
 	return canmove
 
+/mob/living/simple_animal/bullet_act(obj/item/projectile/P)
+	if(!P)
+		return
+	if(prob(mob_reflect_chance))
+		visible_message("<span class='danger'>The [P.name] gets reflected by [name]!</span>")
```

Comment:
I'm pretty sure there is a cleaner way to do this, try looking at reflective armor